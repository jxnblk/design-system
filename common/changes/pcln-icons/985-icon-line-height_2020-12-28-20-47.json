{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Set line-height to 1 to fix issues when aligning icons with Text",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons",
  "email": "email@craigp.me"
}