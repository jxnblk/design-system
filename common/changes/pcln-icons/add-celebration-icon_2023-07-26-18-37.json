{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "add celebration icon",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}