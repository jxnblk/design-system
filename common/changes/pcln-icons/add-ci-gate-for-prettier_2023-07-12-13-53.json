{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Fix formatting with Prettier",
      "type": "none"
    }
  ],
  "packageName": "pcln-icons"
}