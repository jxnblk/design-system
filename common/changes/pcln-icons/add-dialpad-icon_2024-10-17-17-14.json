{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "feat(add-dialpad-icon): add new dialpad icon",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}