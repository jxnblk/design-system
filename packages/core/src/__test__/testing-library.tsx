import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
