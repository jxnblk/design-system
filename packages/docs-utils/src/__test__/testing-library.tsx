import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
