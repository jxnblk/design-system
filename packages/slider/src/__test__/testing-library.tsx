import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
