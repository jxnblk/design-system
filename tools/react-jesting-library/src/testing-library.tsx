import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
