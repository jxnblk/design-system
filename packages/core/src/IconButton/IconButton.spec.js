import React from 'react'
import { render, fireEvent } from 'testing-library'

import { IconButton } from '..'
import { Key as KeyIcon } from 'pcln-icons'

describe('IconButton', () => {
  test('executes onClick prop on click', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(<IconButton icon={<KeyIcon />} onClick={handleClick} />)

    fireEvent.click(getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })

  test('renders without background color in disabled state', () => {
    const { container } = render(<IconButton icon={<KeyIcon />} disabled />)

    const button = container.querySelector('button:disabled')
    expect(button).toHaveStyleRule('background-color', 'transparent')
  })

  test('renders without props', () => {
    const json = rendererCreateWithTheme(<IconButton icon={<KeyIcon />} />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
