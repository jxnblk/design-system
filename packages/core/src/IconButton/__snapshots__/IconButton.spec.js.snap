// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders without props 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  padding: 0;
  background-color: transparent;
  color: inherit;
}

.c0:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c0:hover {
  background-color: transparent;
  color: inherit;
}

.c0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<button
  className="c0"
  name="Key"
>
  <div />
</button>
`;
